engine/gui/src/gui.cpp

```diff
@@ -540,11 +565,15 @@ namespace dmGui
             return RESULT_INVAL_ERROR;
         }
 
-        if (DynamicTexture* t = scene->m_DynamicTextures.Get(texture_hash)) {
-            if (t->m_Deleted) {
+        if (DynamicTexture* t = scene->m_DynamicTextures.Get(texture_hash))
+        {
+            if (t->m_Deleted)
+            {
                 t->m_Deleted = 0;
-                return RESULT_OK;
-            } else {
+                return MakeDynamicTextureData(t, width, height, type, flip, buffer, buffer_size);
+            }
+            else
+            {
```

Comment:
This is the actual bug:
When a texture exists we un-mark it for deletion here so we can reuse it within the frame, but the buffer with the data has already been deleted. The new data is never put into the buffer, and the old buffer is already deleted by this point so the existing texture will essentially be blank.